### ISSUE TITLE
how do you PromotionPieceType

### ISSUE THREAD

**Issue by BryceRobinson0** [SebLague/Chess-Challenge#199]
I what to make any   PromotionPieceType a queen

**Comment by mcthouacbb** [comment]
Can you elaborate and be a bit more specific on the question?

**Comment by JoaoMariaSantos** [comment]
Check out #89

**Comment by BryceRobinson0** [comment]
yay you know in chess when your pawn get  Promotion I what to make it all was be a queen

**Comment by QuinnCG** [comment]
> yay you know in chess when your pawn get Promotion I what to make it all was be a queen

If you mean you want to always promote to queen then I believe out of all legal moves, every possible promotion will be its own move so you can just discard any moves that promote to something other than queen. Haven't tested that but saw someone else give a similar suggestion.

Hope it helps.

**Comment by StevenChristy** [comment]
> yay you know in chess when your pawn get Promotion I what to make it all was be a queen

Yes for most levels of chess play that is true, however, there are instances where promotion to queen can result in a draw by stalemate, or where promotion to another piece can result in a Forced Mate in N moves. The usefulness of non-queen promotions is uncommon, but possible.

**Comment by RedBedHed** [comment]
Yep. Good engines will underpromote when: 

a) The promotion piece is sure to be taken on the next turn (it doesn't really matter in this case). 
b) Underpromotion leads to faster check (maybe promoting to knight puts the king in check etc...).

Your bot needs a search algorithm to achieve this level of awareness.

**Comment by RedBedHed** [comment]
Or a neural network that can predict what a search would find.
